enumerables.rb

```diff
@@ -0,0 +1,66 @@
+# enumerable.rb
+module Enumerable
+  def my_each
+    i = 0
+    while to_a.length > i
+      yield(to_a[i])
+      i += 1
+    end
+  end
```

Comment:
- [ ] enumerables #my_each returns an Enumerator if no block is given
     Failure/Error: expect(array.my_each).to be_an(Enumerator)

- [ ] enumerables #my_each when a block is given when self is an array returns the array itself after calling the given block once for each element in self
     Failure/Error: expect(array.my_each(&block)).to eq(array.each(&block))

- [ ] enumerables #my_each when a block is given when self is a range returns the range itself after calling the given block once for each element in self
     Failure/Error: expect(range.my_each(&block)).to eq(range.each(&block))

- [ ] enumerables #my_each_with_index returns an enumerator if no block is given
     Failure/Error: expect(array.my_each_with_index).to be_an(Enumerator)

Reply:
Solved all the requested changes